tools/codegen/selective_build/operator.py

```diff
@@ -0,0 +1,159 @@
+    # root operator if it is called directly from any one of the models
+    # that this instance of the pytorch library was built for. Hence, it
+    # may not be a root operator in all of the models that are used in
+    # this instance of the pytorch library.
+    is_root_operator: bool
+
+    # Is this operator used for on-device training? If True, then we need to
+    # use the information to generate code in VariableType_N.cpp for registration
+    # of training related operators. Again, this is True if this operator
+    # is used for training in one or more models used by this instance of the
+    # pytorch library.
+    is_used_for_training: bool
+
+    # If True, it indicates that this operator instance (object) refers to an
+    # operator without the overload name and should apply to all overloads
+    # which have this operator name as the base name. This flag is applicable
+    # only for objects that have operator names without a DOT (period) character
+    # in them.
+    #
+    # Note: This flag is a temporary workaround to grandfather in the current
+    # static selective (custom) build mechanism, which largely ignores overload
+    # names when determining whether to select operators for registration
+    # purposes.
+    include_all_overloads: bool
+
+    # Debug Information at the operator level
+    _debug_info: Optional[Tuple[str, ...]]
+
+    @staticmethod
+    def from_yaml_dict(op_name: str, op_info: Dict[str, object]) -> 'SelectiveBuildOperator':
+        allowed_keys = {'name', 'is_root_operator', 'is_used_for_training', 'include_all_overloads', 'debug_info'}
+
+        if len(set(op_info.keys()) - allowed_keys) > 0:
+            raise Exception("Got unexpected top level keys: {}".format(
+                ",".join(set(op_info.keys()) - allowed_keys),
+            ))
+
+        if 'name' in op_info:
+            assert op_name == op_info['name']
+
+        is_root_operator = op_info.get('is_root_operator', True)
+        assert isinstance(is_root_operator, bool)
+
+        is_used_for_training = op_info.get('is_used_for_training', True)
+        assert isinstance(is_used_for_training, bool)
+
+        include_all_overloads = op_info.get('include_all_overloads', True)
+        assert isinstance(include_all_overloads, bool)
+
+        debug_info: Optional[Tuple[str, ...]] = None
+        if 'debug_info' in op_info:
+            di_list = op_info['debug_info']
+            assert isinstance(di_list, list)
+            debug_info = tuple(map(lambda x: str(x), di_list))
+
+        return SelectiveBuildOperator(
+            op_name,
+            is_root_operator,
+            is_used_for_training,
+            include_all_overloads,
+            debug_info,
+        )
+
+    @staticmethod
+    def from_legacy_operator_name_without_overload(name: str) -> 'SelectiveBuildOperator':
+        return SelectiveBuildOperator(
+            name,
+            True,
+            True,
+            True,
+            None,
```

Comment:
Thanks for pointing this out - in fact I think that all call-sites should pass in parameter names to prevent against bugs in case 2 boolean members are switched around. i.e. if that happens, mypy won't complain, but it will be a logical error.